### ISSUE TITLE
Include an __eq__ method for the discord.Embed class

### ISSUE THREAD

**Issue by dob9601** [Rapptz/discord.py#5961]
## The Problem

The discord.Embed class could do with an \_\_eq\_\_ method. Because Discord strips whitespace, the 2 embeds:
```
{'footer': {'text': 'Hint: If your game name/description is multiple characters, wrap it in ""'}, 'fields': [{'inline': False, 'name': 'Syntax: ', 'value': '!lfg <description> <game> <user_count> [interserver] [create_channel]'}
```
and
```
{'footer': {'text': 'Hint: If your game name/description is multiple characters, wrap it in ""'}, 'fields': [{'inline': False, 'name': 'Syntax:', 'value': '!lfg <description> <game> <user_count> [interserver] [create_channel]'}
```
do not evaluate as being equal due to one of the fields in the first embed being "Syntax: " but "Syntax:" in the second

## The Ideal Solution

the discord.Embed class should include an equals method that strips whitespace from the end of each field before comparing the 2 objects

## The Current Solution

Current solution to this is to make sure theres no trailing whitespace in any of your embed fields

## Summary

TL;DR \_\_eq\_\_ function in Embed class to handle whitespace

**Comment by Rapptz** [comment]
This is both computationally expensive (equality checks are meant to be cheap, at most linear in time) and leaks and relies on discord implementation details. I'm afraid I'll have to deny this feature request as-is. Sorry.

**Comment by dob9601** [comment]
That's fair enough. As a sidenote, I've tested this whilst removing the whitespace and the 2 embeds still don't evaluate as being equal. An __eq__ method might still be necessary, just not for the reason I mentioned @Rapptz

**Comment by dob9601** [comment]
I'll move this to a new feature request since its technically a different issue
